lib/common/tests/rules/pcmk__evaluate_attr_expression_test.c

```diff
@@ -0,0 +1,718 @@
+        PCMK_XA_ATTRIBUTE "='num' "                     \
+        PCMK_XA_VALUE "='2' />"
+
+static void
+op_lt_fails(void **state)
+{
+    assert_attr_expression(EXPR_OP_LT_FAILS, pcmk_rc_after_range);
+}
+
+#define EXPR_OP_GT_PASSES                               \
+        "<" PCMK_XE_EXPRESSION " " PCMK_XA_ID "='e' "   \
+        PCMK_XA_TYPE "='" PCMK_VALUE_INTEGER "' "       \
+        PCMK_XA_OPERATION "='" PCMK_VALUE_GT "' "       \
+        PCMK_XA_ATTRIBUTE "='num' "                     \
+        PCMK_XA_VALUE "='2' />"
+
+static void
+op_gt_passes(void **state)
+{
+    assert_attr_expression(EXPR_OP_GT_PASSES, pcmk_rc_ok);
+}
+
+#define EXPR_OP_GT_FAILS                                \
+        "<" PCMK_XE_EXPRESSION " " PCMK_XA_ID "='e' "   \
+        PCMK_XA_TYPE "='" PCMK_VALUE_INTEGER "' "       \
+        PCMK_XA_OPERATION "='" PCMK_VALUE_GT "' "       \
+        PCMK_XA_ATTRIBUTE "='num' "                     \
+        PCMK_XA_VALUE "='20' />"
+
+static void
+op_gt_fails(void **state)
+{
+    assert_attr_expression(EXPR_OP_GT_FAILS, pcmk_rc_before_range);
+}
+
+#define EXPR_OP_LTE_PASSES                              \
+        "<" PCMK_XE_EXPRESSION " " PCMK_XA_ID "='e' "   \
+        PCMK_XA_TYPE "='" PCMK_VALUE_INTEGER "' "       \
+        PCMK_XA_OPERATION "='" PCMK_VALUE_LTE "' "      \
+        PCMK_XA_ATTRIBUTE "='num' "                     \
+        PCMK_XA_VALUE "='10' />"
+
+static void
+op_lte_passes(void **state)
+{
+    assert_attr_expression(EXPR_OP_LTE_PASSES, pcmk_rc_ok);
+}
+
+#define EXPR_OP_LTE_FAILS                               \
+        "<" PCMK_XE_EXPRESSION " " PCMK_XA_ID "='e' "   \
+        PCMK_XA_TYPE "='" PCMK_VALUE_INTEGER "' "       \
+        PCMK_XA_OPERATION "='" PCMK_VALUE_LTE "' "      \
+        PCMK_XA_ATTRIBUTE "='num' "                     \
+        PCMK_XA_VALUE "='9' />"
+
+static void
+op_lte_fails(void **state)
+{
+    assert_attr_expression(EXPR_OP_LTE_FAILS, pcmk_rc_after_range);
+}
+
+#define EXPR_OP_GTE_PASSES                              \
+        "<" PCMK_XE_EXPRESSION " " PCMK_XA_ID "='e' "   \
+        PCMK_XA_TYPE "='" PCMK_VALUE_INTEGER "' "       \
+        PCMK_XA_OPERATION "='" PCMK_VALUE_GTE "' "      \
+        PCMK_XA_ATTRIBUTE "='num' "                     \
+        PCMK_XA_VALUE "='10' />"
+
+static void
+op_gte_passes(void **state)
+{
+    assert_attr_expression(EXPR_OP_GTE_PASSES, pcmk_rc_ok);
+}
+
+#define EXPR_OP_GTE_FAILS                               \
+        "<" PCMK_XE_EXPRESSION " " PCMK_XA_ID "='e' "   \
+        PCMK_XA_TYPE "='" PCMK_VALUE_INTEGER "' "       \
+        PCMK_XA_OPERATION "='" PCMK_VALUE_GTE "' "      \
+        PCMK_XA_ATTRIBUTE "='num' "                     \
+        PCMK_XA_VALUE "='11' />"
+
+static void
+op_gte_fails(void **state)
+{
+    assert_attr_expression(EXPR_OP_GTE_FAILS, pcmk_rc_before_range);
+}
+
+// This also tests that string is used if values aren't parseable as numbers
```

Comment:
Do we want a similar test for integer? The `op_eq_fails` test has `PCMK_VALUE_INTEGER`, but we don't know if the `pcmk_rc_op_unsatisfied` is because the strings are different or because "bar" fails to parse as an integer.

Reply:
meh, seems overkill